pkg/controller/dataimportcron-controller.go

```diff
@@ -1091,6 +1098,90 @@ func addDataImportCronControllerWatches(mgr manager.Manager, c controller.Contro
 	return nil
 }
 
+// addDefaultStorageClassUpdateWatch watches for default/virt default storage class updates
+func addDefaultStorageClassUpdateWatch(mgr manager.Manager, c controller.Controller) error {
+	if err := c.Watch(source.Kind(mgr.GetCache(), &storagev1.StorageClass{}),
+		handler.EnqueueRequestsFromMapFunc(
+			func(ctx context.Context, obj client.Object) []reconcile.Request {
+				log := c.GetLogger().WithName("DefaultStorageClassUpdateWatch")
+				log.Info("Update", "sc", obj.GetName(),
+					"default", obj.GetAnnotations()[cc.AnnDefaultStorageClass] == "true",
+					"defaultVirt", obj.GetAnnotations()[cc.AnnDefaultVirtStorageClass] == "true")
+				reqs, err := getReconcileRequestsForDicsWithPendingPvc(ctx, mgr.GetClient())
+				if err != nil {
+					log.Error(err, "Failed getting DataImportCrons with pending PVCs")
+				}
+				return reqs
+			},
+		),
+		predicate.Funcs{
+			CreateFunc: func(event.CreateEvent) bool { return false },
+			DeleteFunc: func(event.DeleteEvent) bool { return false },
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				scOld := e.ObjectOld.(*storagev1.StorageClass)
+				scNew := e.ObjectNew.(*storagev1.StorageClass)
+				return (scNew.Annotations[cc.AnnDefaultStorageClass] != scOld.Annotations[cc.AnnDefaultStorageClass]) ||
```

Comment:
Not sure if this is correct. From the [kubernetes documentation](https://kubernetes.io/docs/concepts/storage/storage-classes/#default-storageclass)
```
If you set the [storageclass.kubernetes.io/is-default-class](https://kubernetes.io/docs/reference/labels-annotations-taints/#storageclass-kubernetes-io-is-default-class) annotation to true on more than one StorageClass in your cluster, and you then create a PersistentVolumeClaim with no storageClassName set, Kubernetes uses the most recently created default StorageClass.
```
So it is possible to have multiple storage classes with the annotation set to true. You just have to picked the most recently created storage class.

Reply:
We will pick the most recently **default-updated** SC, but anyway we reconcile all the relevant DICs, and decide whether to delete the PVCs based on the existing default k8s/virt SC logic.

In case of SC creation, we are already covered by k8s as mentioned in the PR description.